### PR TITLE
docs(recipes): add recipe for automock library

### PR DESCRIPTION
**Pull request by omermorad**
## PR Checklist
Please check if your PR fulfills the following requirements:

- [x] The commit message follows our guidelines: https://github.com/nestjs/docs.nestjs.com/blob/master/CONTRIBUTING.md

## PR Type
What kind of change does this PR introduce?
- [x] Docs

**Issue Number: #2288** 

Automock in GitHub: https://github.com/omermorad/automock
NPM: https://www.npmjs.com/package/@automock/jest

### CODE REVIEW

**Comment by omermorad** [comment]
@jmcdo29 @kamilmysliwiec I'm still working on this one, I have tons of work in my workplace, but I believe I will finish this one soon.
@kamilmysliwiec in the meanwhile, I would really appreciate your opinion in this thread https://github.com/nestjs/docs.nestjs.com/issues/2288 :) :)

**Comment by omermorad** [comment]
@kamilmysliwiec, @jmcdo29 - I think everything is ready, I will appreciate your review :)

File: content/recipes/automock.md [hunk 1]
```diff
@@ -0,0 +1,162 @@
...
+
+The dependency injection (DI) container is an essential component of the Nest module system.
+This container is utilized both during testing, and the application execution. \
+Unit tests vary from other types of tests, such as integration tests, in that they must
+fully override providers/services within the DI container. External class dependencies
+(providers) of the so-called "unit", have to be totally isolated. That is, all dependencies
+within the DI container should be replaced by mock objects. \
+As a result, loading the target module and replacing the providers inside it is a process
+that loops back on itself. Automock tackles this issue by automatically mocking all the
+class external providers, resulting in total isolation of the unit under test.
```
Comment (jmcdo29) [inline content/recipes/automock.md:17]:
```suggestion
The dependency injection (DI) container is an essential component of the Nest module system.
This container is utilized both during testing, and the application execution.
Unit tests vary from other types of tests, such as integration tests, in that they must
fully override providers/services within the DI container. External class dependencies
(providers) of the so-called "unit", have to be totally isolated. That is, all dependencies
within the DI container should be replaced by mock objects.
As a result, loading the target module and replacing the providers inside it is a process
that loops back on itself. Automock tackles this issue by automatically mocking all the
class external providers, resulting in total isolation of the unit under test.
```

File: content/recipes/automock.md [hunk 2]
```diff
@@ -0,0 +1,162 @@
+### Automock
+
+Automock is a standalone library for unit testing. Using TypeScript Reflection
+API (`reflect-metadata`) internally to produce mock objects, Automock streamlines
+test development by automatically mocking class external dependencies.
+
```
Comment (jmcdo29) [inline content/recipes/automock.md:6]:
```suggestion

> info **info** `automock` is a third party package and is not managed by the NestJS core team. Please, report any issues found with the library in the [appropriate repository](<add your repo here>)

```

File: content/recipes/automock.md [hunk 3]
```diff
@@ -0,0 +1,162 @@
...
+```
+
+Automock does not require any additional setup.
+
+> info **info** Jest is the only test framework currently supported by Automock.
+Sinon will shortly be released.
+
+#### Example
+Consider the following cats service, which takes three constructor parameters:
+```ts
```
Comment (jmcdo29) [inline content/recipes/automock.md:32]:
```suggestion
#### Example

Consider the following cats service, which takes three constructor parameters:

```ts
```
added whitespace

File: content/recipes/automock.md [hunk 4]
```diff
@@ -0,0 +1,162 @@
...
+    });
+  });
+});
+```
+
+> info **info** The jest.Mocked<Source> utility type returns the Source type
+> wrapped with type definitions of Jest mock function. ([reference](https://jestjs.io/docs/mock-function-api/#jestmockedsource))
+
+#### About `unit` and `unitRef`
+Let's examine the following code:
```
Comment (jmcdo29) [inline content/recipes/automock.md:95]:
```suggestion
#### About `unit` and `unitRef`

Let's examine the following code:
```

**Review by jmcdo29 — CHANGES_REQUESTED** [review]
Looking good. I noticed a few small changes to make. One question, what does the `.mock()` do? Maybe add that in with the `unit` and `unitRef` section as it all comes around `.compile()`?

File: content/recipes/automock.md (hunk 2 above)
Comment (omermorad) [reply content/recipes/automock.md:8]: Added manually.

**Review by jmcdo29 — APPROVED** [review]

**Review by Tony133 — APPROVED** [review]

File: content/recipes/automock.md [hunk 5]
```diff
@@ -0,0 +1,167 @@
...
+export class CatsService {
+  constructor(
+    private logger: Logger,
+    private httpService: HttpService,
+    private catsDal: CatsDal,
+  ) {}
+
+  async getAllCats() {
+    const cats = await this.httpService.get('http://localhost:3000/api/cats');
+    this.logger.log('successfully fetched all cats');
```
Comment (kamilmysliwiec) [inline content/recipes/automock.md:50]:
```suggestion
    this.logger.log('Successfully fetched all cats');
```

**Comment by kamilmysliwiec** [comment]
lgtm

**Comment by kamilmysliwiec** [comment]
lgtm

**Pull request merged**